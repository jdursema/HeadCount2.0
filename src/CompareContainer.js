import React from 'react';
import Card from './Card';

const CompareContainer = ({selectedArray, changeClass}) => {
  const mappedSectedCards= selectedArray.map((selectedCard)=>{
    return <Card key={selectedCard.location} 
                     location={selectedCard.location} 
                     data={selectedCard.data}
                     changeClass={changeClass}
                     type = 'card'
                     />
  })
  return(
    <div>

      {mappedSectedCards}
    </div>
  )
}

export default CompareContainer;
