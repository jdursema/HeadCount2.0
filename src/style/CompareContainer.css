.compared-cards-section{
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  height: 475px;
}

.compare-card{
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 305px;
}

.comparison{
  font-size: 100px;
}

.average{
  font-size: 25px;
}