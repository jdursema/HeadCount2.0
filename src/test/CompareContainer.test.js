import React from 'react';
import { shallow, mount } from 'enzyme';
import CompareContainer from '../CompareContainer';


describe('CompareContainer ', () => {

  it('should be defined', () => {
    const mockFunc = jest.fn();
    const compareContainer = shallow(<CompareContainer selectedArray={[]}changeClass={mockFunc} />)
    expect(compareContainer).toBeDefined();
  });

  it('should receive props', () => {
    const mockFunc = jest.fn();
    const compareContainer = shallow(<CompareContainer selectedArray={[]}changeClass={mockFunc} />)

    expect(compareContainer.instance().props.selectedArray).toEqual([])
    expect(compareContainer.instance().props.changeClass).toEqual(mockFunc)
  });

  it('should match snap shot of compare container' , () =>{
    const mockFunc = jest.fn();
    let compareContainer = shallow(<CompareContainer selectedArray={[]} changeClass={mockFunc} compareCard={mockFunc}/>)
    compareContainer = shallow(<CompareContainer selectedArray={[{location: 'COLORADO', data: {2001: .5} }]}changeClass={mockFunc} compareCard={mockFunc}/>)

    expect(compareContainer).toMatchSnapshot();

  })

});
