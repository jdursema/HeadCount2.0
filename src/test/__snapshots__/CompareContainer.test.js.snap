exports[`CompareContainer  should match snap shot of compare container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    changeClass={[Function]}
    compareCard={[Function]}
    selectedArray={
        Array [
            Object {
              "data": Object {
                "2001": 0.5,
              },
              "location": "COLORADO",
            },
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          changeClass={[Function]}
          data={
                    Object {
                              "2001": 0.5,
                            }
          }
          location="COLORADO"
          type="card" />,
        <div />,
        undefined,
      ],
      "className": "compared-cards-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "COLORADO",
        "nodeType": "function",
        "props": Object {
          "changeClass": [Function],
          "data": Object {
            "2001": 0.5,
          },
          "location": "COLORADO",
          "type": "card",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            changeClass={[Function]}
            data={
                        Object {
                                    "2001": 0.5,
                                  }
            }
            location="COLORADO"
            type="card" />,
          <div />,
          undefined,
        ],
        "className": "compared-cards-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "COLORADO",
          "nodeType": "function",
          "props": Object {
            "changeClass": [Function],
            "data": Object {
              "2001": 0.5,
            },
            "location": "COLORADO",
            "type": "card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
